src/one_wire.toit

```diff
@@ -85,26 +78,46 @@ class Protocol:
       result[it] = decode_signals_to_bits_ signals --from=write_signal_count + it * SIGNALS_PER_BYTE
     return result
 
-  static encode_read_signals_ signals/rmt.Signals --from/int=0 --bit_count/int:
-    assert: 0 <= from
-    assert: from + bit_count * SIGNALS_PER_BIT <= signals.size
+  static encode_read_signals_ --bit_count/int -> rmt.Signals:
+    signals := rmt.Signals (bit_count * SIGNALS_PER_BIT)
     bit_count.repeat:
-      i := from + it * SIGNALS_PER_BIT
-      signals.set_signal i READ_INIT_TIME_STD 0
-      signals.set_signal i + 1 IO_TIME_SLOT - READ_INIT_TIME_STD 1
+      i := it * SIGNALS_PER_BIT
+      signals.set i --period=READ_INIT_TIME_STD --level=0
+      signals.set (i + 1) --period=(IO_TIME_SLOT - READ_INIT_TIME_STD) --level=1
+    return signals
 
   /**
   Writes $count bits from $value to the receiver.
   */
   write_bits value/int count/int -> none:
-    signals :=  rmt.Signals count * SIGNALS_PER_BIT
-    encode_write_signals_ signals value --count=count
+    signals := encode_write_signals_ value --count=count
     rmt.transmit tx_channel_ signals
 
   write_byte value/int -> none:
     write_bits value BITS_PER_BYTE
 
-  static encode_write_signals_ signals/rmt.Signals bits/int --from/int=0 --count/int=8 -> none:
+  /**
+  Encodes the given integer or byte array to a sequence of signals.
+
+  The $bits_or_bytes must be either an integer, in which case the $count must be given.
+  If $bits_or_bytes is a byte array, then the $count must be equal to $bits_or_bytes * 8.
+  */
+  static encode_write_signals_ bits_or_bytes/any --count/int -> rmt.Signals:
+    // Add one bit for the termination signal.
```

Comment:
Where is this added?

Reply:
oops.
Now done by the RMT library and not here anymore.